app/src/main/java/lt/gyvosistorijos/service/GeofenceIntentService.kt

```diff
@@ -66,13 +64,15 @@ class GeofenceIntentService : IntentService(GeofenceIntentService.TAG) {
         )
 
 
-        val triggeredStory = StoryDb.getById(triggeringGeofencesIdsList.first())
 
-        if (triggeredStory != null) {
-            sendNotification(triggeredStory)
-        } else {
-            Timber.w("Triggered story is null. " +
-                    "TriggeringGeofencesIdsList = $triggeringGeofencesIdsList}")
+        when (geofenceTransition) {
+            Geofence.GEOFENCE_TRANSITION_ENTER ->
+                StoryDb.getById(triggeringGeofencesIdsList.first())?.let { story ->
+                    sendNotification(story)
+                }
+            Geofence.GEOFENCE_TRANSITION_EXIT ->
+                triggeringGeofencesIdsList.map { StoryDb.getById(it) }
```

Comment:
It's not necessary to do a DB fetch to act on Story objects if you're just dealing directly with the ids here anyway.